libopae/plugins/xfpga/sysfs.c

```diff
@@ -1173,22 +1173,66 @@ fpga_result get_port_sysfs(fpga_handle handle, char *sysfs_port)
 	return FPGA_OK;
 }
 
-// get fpga device id
-fpga_result get_fpga_deviceid(fpga_handle handle, uint64_t *deviceid)
+enum fpga_hw_type opae_id_to_hw_type(uint16_t vendor_id, uint16_t device_id)
+{
+	enum fpga_hw_type hw_type = FPGA_HW_UNKNOWN;
+
+	if (vendor_id == 0x8086) {
+
+		switch (device_id) {
+		case 0xbcbc:
+		case 0xbcbd:
+		case 0xbcbe:
+		case 0xbcbf:
+		case 0xbcc0:
+		case 0xbcc1:
+		case 0x09cb:
+			hw_type = FPGA_HW_MCP;
+		break;
+
+		case 0x09c4:
+		case 0x09c5:
+			hw_type = FPGA_HW_DCP_RC;
+		break;
+
+		case 0x0b2b:
+		case 0x0b2c:
+			hw_type = FPGA_HW_DCP_DC;
+		break;
+
+		case 0x0b30:
+		case 0x0b31:
+			hw_type = FPGA_HW_DCP_VC;
+		break;
+
+		default:
+			FPGA_ERR("unknown device id: 0x%04x", device_id);
+		}
+
+	} else {
+		FPGA_ERR("unknown vendor id: 0x%04x", vendor_id);
+	}
```

Comment:
Should we define constants/macros for these ids?

Reply:
I'd prefer to only track them here, and use fpga_hw_type otherwise.  I was hoping to remove the notion of referring directly to device id's in the code.